src/adam7.rs

```diff
@@ -233,6 +266,145 @@ pub fn expand_pass(
+// pixel value without leaving any pixels empty, helping to produce smoother
+// animation for interlaced images.
+// For example, after the 3rd pass using expand_pass, the image for an 8x8 size
+// would look like this:
+//   1 - - - 2 - - -
+//   - - - - - - - -
+//   - - - - - - - -
+//   - - - - - - - -
+//   3 - - - 3 - - -
+//   - - - - - - - -
+//   - - - - - - - -
+//   - - - - - - - -
+// On the other hand, after the 3rd pass using splat_expand_pass, the image
+// would look like this:
+//   1 1 1 1 2 2 2 2
+//   1 1 1 1 2 2 2 2
+//   1 1 1 1 2 2 2 2
+//   1 1 1 1 2 2 2 2
+//   3 3 3 3 3 3 3 3
+//   3 3 3 3 3 3 3 3
+//   3 3 3 3 3 3 3 3
+//   3 3 3 3 3 3 3 3
+pub fn splat_expand_pass(
+    img: &mut [u8],
+    img_row_stride: usize,
+    interlaced_row: &[u8],
+    interlace_info: &Adam7Info,
+    bits_per_pixel: u8,
+    frame_width_in_pixels: usize,
+) {
+    let bits_pp = bits_per_pixel as usize;
+
+    let bit_indices = expand_adam7_bits(img_row_stride, interlace_info, bits_pp);
+
+    if bits_pp < 8 {
+        for (pos, px) in bit_indices.zip(subbyte_pixels(interlaced_row, bits_pp)) {
+            bit_splat_expand(
+                img,
+                img_row_stride * 8,
+                px,
+                interlace_info,
+                bits_pp,
+                pos,
+                frame_width_in_pixels * bits_pp,
+            );
+        }
+    } else {
+        let bytes_pp = bits_pp / 8;
+
+        for (bitpos, px) in bit_indices.zip(interlaced_row.chunks(bytes_pp)) {
+            let start_byte = bitpos / 8;
+            byte_splat_expand(
+                img,
+                img_row_stride,
+                px,
+                interlace_info,
+                bytes_pp,
+                start_byte,
+            );
+        }
+    }
+}
+
+// TODO: Eliminate the duplicate code in |bit_splat_expand| and
+// |byte_splat_expand|, if possible.
+fn bit_splat_expand(
+    img: &mut [u8],
+    stride_in_bits: usize,
+    px: u8,
+    info: &Adam7Info,
+    bits_pp: usize,
+    bit_pos: usize,
+    frame_width_in_bits: usize,
+) {
+    assert!(
+        frame_width_in_bits <= stride_in_bits,
+        "The frame width should be less than or equal to the row length."
+    );
+    let height = (img.len() * 8 - bit_pos + stride_in_bits - 1) / stride_in_bits;
+    let pass_const = &PASS_CONSTANTS[info.pass as usize - 1];
+    let (x_repeat, y_repeat) = (pass_const.x_repeat(), pass_const.y_repeat());
+    for i in 0..min(y_repeat, height) {
+        let offset = bit_pos + i * stride_in_bits;
+        let max_fill = min(
+            (frame_width_in_bits - bit_pos % stride_in_bits) / bits_pp,
+            x_repeat,
+        );
+        copy_nbits_mtimes(img, offset, bits_pp, px, max_fill);
+    }
+}
+
+fn byte_splat_expand(
+    img: &mut [u8],
+    stride: usize,
+    px: &[u8],
+    info: &Adam7Info,
+    bytes_pp: usize,
+    start_byte: usize,
+) {
+    let height = (img.len() - start_byte + stride - 1) / stride;
```

Comment:
I think this can use the [`div_ceil`](https://doc.rust-lang.org/std/primitive.usize.html#method.div_ceil) method